gradle/scripts/lib/java.gradle

```diff
@@ -91,8 +122,12 @@ configure(projectsWithFlags('java')) {
 
     // Set the sensible compiler options.
     tasks.withType(JavaCompile) {
-        sourceCompatibility = project.findProperty('javaSourceCompatibility') ?: '1.8'
-        targetCompatibility = project.findProperty('javaTargetCompatibility') ?: '1.8'
+        sourceCompatibility = project.findProperty('javaSourceCompatibility') ?: "${targetJavaVersion}"
```

Comment:
Does this work when it's JDK 1.8?
previously it was `1.8` but now it's `8`.

Reply:
I believe the following method is called before usage internally

https://github.com/gradle/gradle/blob/5ec3f672ed600a86280be490395d70b7bc634862/subprojects/base-services/src/main/java/org/gradle/api/JavaVersion.java#L139

I also found unit tests supporting this
https://github.com/gradle/gradle/blob/585397da80852998840f19638057c90df2c8302a/subprojects/java-compiler-plugin/build.gradle.kts#L10

Reply:
I didn't know that Gradle covers both cases. Thanks! 😄

Reply:
Let me just pass a `JavaVersion` instead of a string to make the intention clearer 😅